supervisor/supervisor.py

```diff
@@ -65,17 +65,17 @@ def need_update(self) -> bool:
             return False
 
         try:
-            return pkg_parse(self.version) < pkg_parse(self.latest_version)
-        except (TypeError, ValueError):
+            return self.version < self.latest_version
+        except AwesomeVersionException:
```

Comment:
```suggestion
        except (AwesomeVersionException, TypeError):
```
In case both sides are `None`